roofit/roofitcore/src/RooProdPdf.cxx

```diff
@@ -610,19 +609,15 @@ void RooProdPdf::factorizeProduct(const RooArgSet& normSet, const RooArgSet& int
     RooArgSet pdfAllDeps; // All dependents of this PDF
 
     // Make list of all dependents of this PDF
-    RooArgSet* tmp = getObservables(normSet);
-    pdfAllDeps.add(*tmp);
-    delete tmp;
+    pdfAllDeps.add(*std::unique_ptr<RooArgSet>{getObservables(normSet)});
```

Comment:
Wow, danger!

- Wrap stuff in unique_ptr
- Dereference it to add to pdfAllDeps (which will use `&object` to store the ptr of the incoming object)
- Delete the object, because pointer goes out of scope?

Oh, wait. It's the set and not the object that's deleted. Ok, that should work. But maybe two lines are a bit nicer for readers.

Reply:
Okay, I'll put it in two lines